### DISCUSSION TITLE
When enabling EVPN L3VNI, also MPLS L3VPN stuff is generated for a VRF

### DISCUSSION THREAD

**Discussion by ssasso** [ipspace/netlab#343] [General]
Reading at https://github.com/ipspace/netsim-tools/blob/e76dfe39453e4e8b1662f321093d0543eca2d167/netsim/ansible/templates/vrf/eos.j2#L1
and https://github.com/ipspace/netsim-tools/blob/e76dfe39453e4e8b1662f321093d0543eca2d167/netsim/ansible/templates/vrf/eos.bgp.j2#L17

it seems that, when a VRF is enabled, all the MPLS L3VPN stuff is enabled by default.

This *may* generate useless chaos when only EVPN L3VNI is used.

I'm trying to adopt this on the VyOS template for *vrf*, don't know if you have better alternatives or ideas:
```
{%   if not 'evpn' in vdata %}
... generate MPLS L3VPN stuff
```

**Comment by ipspace** [comment]
The reason I have it there is route leaking. You need RT and route redistribution into BGP for inter-VRF route leaking even if you don't 
run MPLS VPN. The rest of the L3VPN stuff (BGP AF) is not enabled in VRF template.

If you're not facing the same challenge, move the MPLS L3VPN stuff into MPLS-related templates.
